src/geometry/lane_geometry.h

```diff
@@ -0,0 +1,107 @@
+// BSD 3-Clause License
+//
+// Copyright (c) 2022, Woven Planet.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+#include <memory>
+
+#include <maliput/common/maliput_copyable.h>
+#include <maliput/math/roll_pitch_yaw.h>
+#include <maliput/math/vector.h>
+
+#include "maliput_sparse/geometry/line_string.h"
+
+namespace maliput_sparse {
+namespace geometry {
+
+class LaneGeometry {
+ public:
+  MALIPUT_NO_COPY_NO_MOVE_NO_ASSIGN(LaneGeometry);
+
+  LaneGeometry(const LineString3d& left, const LineString3d& right, double scale_length, double linear_tolerance);
+  ~LaneGeometry() = default;
+
+  double p0() const { return 0.; }
+  double p1() const { return centerline_.length(); }
+
+  /// @return The arc length of the centerline of the lane.
+  double ArcLength();
+
+  /// @return The linear tolerance used to compute all the methods.
+  /// @see maliput::api::RoadGeometry::linear_tolerance().
+  double linear_tolerance() const { return linear_tolerance_; }
+
+  /// @return The scale length of the tolerance used to compute all the methods.
+  /// @see maliput::api::RoadGeometry::scale_length().
+  double scale_length() const { return scale_length_; }
+
+  /// Evaluates @f$ W(p, r, h) @f$.
+  ///
+  /// @param prh A vector in the LaneGeometry domain.
+  /// @return A vector in the INERTIAL Frame which is the image of the LaneGeometry.
+  maliput::math::Vector3 W(const maliput::math::Vector3& prh) const;
```

Comment:
FTR: The `p` coordinate isn't at ground level but along the 3d-centerline. It differs with `maliput_malidrive`'s implementation where we had `p` as a coordinate in the ground curve.